### ISSUE TITLE
Running on ROS Kinetic

### ISSUE THREAD

**Issue by afsyaw** [SteveMacenski/cyton_gamma_300-1500_operation_and_simulation#6]
Tried running the recent update and had this error.

`[ INFO] [1483046237.591502060, 0.356000000]: Loaded gazebo_ros_control.
[ WARN] [1483046237.594396310, 0.356000000]: GazeboRosJointStatePublisher Plugin (ns = //) missing <updateRate>, defaults to 100.000000
[ INFO] [1483046237.594455031, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: shoulder_roll_joint
[ INFO] [1483046237.594469785, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: shoulder_pitch_joint
[ INFO] [1483046237.594483314, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: shoulder_yaw_joint
[ INFO] [1483046237.594499425, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: elbow_pitch_joint
[ INFO] [1483046237.594514194, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: elbow_yaw_joint
[ INFO] [1483046237.594530818, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: wrist_pitch_joint
[ INFO] [1483046237.594543048, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: wrist_roll_joint
[ INFO] [1483046237.594551832, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: gripper_joint
[ INFO] [1483046237.594559180, 0.356000000]: GazeboRosJointStatePublisher is going to publish joint: gripper_joint2
[ INFO] [1483046237.594583570, 0.356000000]: Starting GazeboRosJointStatePublisher Plugin (ns = //)!, parent name: cyton_gamma_1500
Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/controller_manager/spawner", line 212, in <module>
    if __name__ == '__main__': main()
  File "/opt/ros/kinetic/lib/controller_manager/spawner", line 190, in main
    resp = load_controller(name)
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/impl/tcpros_service.py", line 435, in __call__
    return self.call(*args, **kwds)
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/impl/tcpros_service.py", line 505, in call
    raise ServiceException("unable to connect to service: %s"%e)
rospy.service.ServiceException: unable to connect to service: [Errno 104] Connection reset by peer
Segmentation fault (core dumped)
[gazebo-2] process has died [pid 10195, exit code 139, cmd /opt/ros/kinetic/lib/gazebo_ros/gzserver -e ode worlds/empty.world __name:=gazebo __log:=/home
`

**Comment by SteveMacenski** [comment]
Hi, can you give me some more information? What commands are you running? I assume the simulation files since this is a gazebo traceback. 

I run Indigo but if the previous release worked for you in Kinetic, I can see if I can get this error. I admittedly don't use the simulation environment much, but I can look into it. It may be something here for hardware makes the simulator unhappy. 

From looking at the trace, it may just be a bad attempt. Sometimes Gazebo is touchy. How many times did you try running this?

**Comment by afsyaw** [comment]
On first running with Kinetic, there were errors stating some headers needed C++11
So I added `set(CMAKE_CXX_FLAGS "-std=gnu++11")` to some of the CMakeLists files which let me compile.
When I tried running `roslaunch cyton_gamma_pkg simulation_gamma_1500.launch` I had the error above among the output.

It persisted past a couple of runs, definitely more than 5.

**Comment by SteveMacenski** [comment]
It looks like you're missing dependencies. 

> sudo apt-get install ros-kinetic-gazebo-ros-pkgs ros-kinetic-gazebo-ros-control

Should do the first step. 

I had some issues with the URDF that I just took care of. I will upload an updated version soon. Fixing the 300 simulation at the same time will take me a bit. 

Edit: I will pull the changes to github on Monday.

**Comment by afsyaw** [comment]
Those dependencies were already installed as I have the previous version running.

Sure, I'd check back then.

**Comment by SteveMacenski** [comment]
Hi, I have fixed this repo for you. I would appreciate if you would tell me everywhere you had to add the C++11 flags so I can update them on my end as well, or if you want to submit a PR. Once that is done, I will start supporting both indigo and kinetic. 

Thanks for helping! Out of curiosity, what are you using it for and with what group?

**Comment by afsyaw** [comment]
Currently, I have the flags in the CMakeLists files for
cyton_gamma_pkg
industrial_core-indigo-devel>industrial_trajectory_filters
moveit_plugins-indigo-devel>moveit_controller_manager_example
			                    >moveit_cyton_controller_manager
			                    >moveit_fake_controller_manager
   			                    >moveit_simple_controller_manager

In trying to figure out an elegant approach, I had some success with adding that line to the top level CMakeLists file for the catkin workspace, and using the kinetic version of the moveit plugins and the jade version of the industrial core. Considering the bug above persisted, I am not sure if both are required or if one will suffice.

**Comment by afsyaw** [comment]
We would be using the arm for dynamic performance tests of search algorithms on mobile manipulators. I'm with the Computer Vision and Sensing Systems (COVISS) Lab at Marquette.

**Comment by SteveMacenski** [comment]
You can delete the industrial core and moveit_plugins, those are just there fore book keeping and don't have any dependency on my work. Just apt-get them for your version, which is sounds like you have. Perhaps its a versioning problem in the kinetic/jade version of those drivers?

I'll add the tag to those packages. 

The same bug is still there? It works fine in my systems. Did your workspace compile fine? 

Thanks for sharing! These robots are in the SSPD office at NASA Goddard and I use them as a tech demonstration for integrating ROS (without moveit) onto our larger robots (Motoman, Yaskawa, Fanuc). I go to UIUC so this is a during breaks gig.

**Comment by afsyaw** [comment]
The issue is gone. I cloned the update, deleted both folders [moveit_plugins-indigo-devel & industrial_core-indigo-devel] and it compiles fine. 
Thanks.

All the best on your work with the SSPD. On orbit servicing is an interesting challenge. It would be nice to see ROS up there again.

**Comment by SteveMacenski** [comment]
Awesome, feel free to open another ticket if you hit any other snags. I'd appreciate a PR on the C++11 changes you implemented for Kinetic to add here.

Thanks! Your work looks really interesting, I'm currently working back at UIUC on perception of robotic manipulation of deformable rods. Mobile-manipulators are a favorite of mine.

**Comment by afsyaw** [comment]
Well I don't think there were any others. After deleting the two folders, it worked without issues. I am currently working the update.

I may make one for the collisions though. I just finished editing those.

I would try to email you on that and a few others before, so as to not pollute this thread.
